File: cinema-backend/src/main/java/org/example/cinemabackend/movie/application/dto/response/ProductionDetailsResponse.java
package org.example.cinemabackend.movie.application.dto.response;

import lombok.Builder;

import java.time.LocalDate;
import java.util.Set;

@Builder
public record ProductionDetailsResponse(
        LocalDate worldPremiereDate,
        FilmMemberResponse director,
        Set<FilmMemberResponse> actors,
        Set<String> originalLanguages,
        Set<String> productionCountries
) {
}

File: cinema-frontend/src/app/movie/dtos/response/production-details.response.ts
import {FilmMemberResponse} from "./film-member.response";

export class ProductionDetailsResponse {
  constructor(
    readonly worldPremiereDate: Date,
    readonly director: FilmMemberResponse,
    readonly actors: FilmMemberResponse[],
    readonly originalLanguages: string[],
    readonly productionCountries: string[],
  ) {
  }

}
